src/zulip/mod.rs

```diff
@@ -19,23 +20,37 @@ impl SyncZulip {
         dry_run: bool,
     ) -> anyhow::Result<Self> {
         let zulip_api = ZulipApi::new(username, token, dry_run);
+        let stream_definitions = get_stream_definitions(team_api, &zulip_api)?;
         let user_group_definitions = get_user_group_definitions(team_api, &zulip_api)?;
         let zulip_controller = ZulipController::new(zulip_api)?;
         Ok(Self {
             zulip_controller,
+            stream_definitions,
             user_group_definitions,
         })
     }
 
     pub(crate) fn diff_all(&self) -> anyhow::Result<Diff> {
-        self.user_group_definitions
+        self.stream_definitions
             .iter()
-            .filter_map(|(user_group_name, member_ids)| {
-                self.diff_user_group(user_group_name, member_ids)
+            .filter_map(|(stream_name, member_ids)| {
+                self.diff_stream_membership(stream_name, member_ids)
                     .transpose()
             })
             .collect::<anyhow::Result<Vec<_>>>()
-            .map(|user_group_diffs| Diff { user_group_diffs })
+            .and_then(|stream_membership_diffs| {
+                self.user_group_definitions
+                    .iter()
+                    .filter_map(|(user_group_name, member_ids)| {
+                        self.diff_user_group(user_group_name, member_ids)
+                            .transpose()
+                    })
+                    .collect::<anyhow::Result<Vec<_>>>()
+                    .map(|user_group_diffs| Diff {
+                        user_group_diffs,
+                        stream_membership_diffs,
+                    })
+            })
```

Comment:
I find this more readable:

```rs
        let stream_membership_diffs = self
            .stream_definitions
            .iter()
            .filter_map(|(stream_name, member_ids)| {
                self.diff_stream_membership(stream_name, member_ids)
                    .transpose()
            })
            .collect::<anyhow::Result<Vec<_>>>()?;
        let user_group_diffs = self
            .user_group_definitions
            .iter()
            .filter_map(|(user_group_name, member_ids)| {
                self.diff_user_group(user_group_name, member_ids)
                    .transpose()
            })
            .collect::<anyhow::Result<Vec<_>>>()?;
        Ok(Diff {
            user_group_diffs,
            stream_membership_diffs,
        })
```

what do you think?

Reply:
Fixed!